api/client/service.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+
+package nighthawk.client;
+
+import "api/client/options.proto";
+import "api/client/output.proto";
+
+message StartRequest {
+  CommandLineOptions options = 1;
+}
+
+/*
+TODO(oschaaf): Not implemented yet.
+*/
+message UpdateRequest {
+}
+
+/*
+TODO(oschaaf): Not implemented yet.
+*/
+message CancellationRequest {
+}
+
+message ExecutionRequest {
+  oneof command_specific_options {
+    StartRequest start_request = 1;
+    UpdateRequest update_request = 2;
+    CancellationRequest cancellation_request = 3;
+  }
+}
+
+/*
+Output is guaranteed to be set when success is true. When success is false,
+output may or may not be set, and error message is guaranteed to be set.
+*/
+message ExecutionResponse {
+  Output output = 1;
+  bool success = 2;
+  string error_message = 3;
+}
+
+service NighthawkService {
+
+  /*
+  Accepts a stream of ExecutionRequests, which can be used to initiate, update, or cancel a test.
+  Returns a stream of ExecutionResponse objects which will contain results for all Nighthawk runs.
+  */
+  rpc sendCommand(stream ExecutionRequest) returns (stream ExecutionResponse) {
```

Comment:
Nit: s/sendCommand/ExecutionStream; the RPC is a series of request/responses around an execution, rather than sending a single command. This can also be reflected in method names below.

Reply:
Fixed, I think, please double check.